src/HasMedia/HasMediaTrait.php

```diff
@@ -94,6 +95,50 @@ public function addMediaFromUrl(string $url)
     }
 
     /**
+     * Add a base64 encoded file to the medialibrary.
+     *
+     * @param string $data
+     *
+     * @throws InvalidBase64Data
+     * @throws \Spatie\MediaLibrary\Exceptions\FileCannotBeAdded
+     *
+     * @return \Spatie\MediaLibrary\FileAdder\FileAdder
+     */
+    public function addMediaFromBase64(string $data)
+    {
+        if (! is_string($data)) {
+            throw InvalidBase64Data::create();
+        }
+
+        // strip out data uri scheme information (see RFC 2397)
+        if (strpos($data, 'base64') !== false) {
+            list(, $data) = explode(';', $data);
+            list(, $data) = explode(',', $data);
+        }
+
+        // strict mode filters for non-base64 alphabet characters
+        if (base64_decode($data, true) === false) {
+            throw InvalidBase64Data::create();
+        }
+
+        // decoding and then reeconding should not change the data
+        if (base64_encode(base64_decode($data)) !== $data) {
+            throw InvalidBase64Data::create();
+        }
+
+        $data = base64_decode($data);
+
+        // temporarily store the decoded data on the filesystem to be able to pass it to the fileAdder
+        $tmpFile = tempnam(sys_get_temp_dir(), 'media-library');
```

Comment:
Rename `media-library` to `medialibrary` (that's a bit more consistent with the reset of the code)